graylog2-web-interface/src/components/common/DataTable.jsx

```diff
@@ -121,36 +132,59 @@ class DataTable extends React.Component {
     this.setState({ filteredRows });
   };
 
+  _updateState() {
```

Comment:
Since `headers` and `rows` always mirror the values in `props`, I think it would be good to remove them from the state. This method should only reset the `filteredRows` if the original `rows` changed. We should also rename the method to make clearer what it does.